apiserver/src/test/java/org/dependencytrack/resources/v1/ProjectResourceTest.java

```diff
@@ -603,18 +634,14 @@ public void getProjectLookupNotPermittedTest() {
                 .request()
                 .header(X_API_KEY, apiKey)
                 .get();
-        assertThat(response.getStatus()).isEqualTo(403);
-        assertThatJson(getPlainTextBody(response)).isEqualTo(/* language=JSON */ """
-                {
-                  "status": 403,
-                  "title": "Project access denied",
-                  "detail": "Access to the requested project is forbidden"
-                }
-                """);
+        assertThat(response.getStatus()).isEqualTo(404);
+        assertThat(getPlainTextBody(response)).isEqualTo("The project could not be found.");
```

Comment:
Note: `preprocessACLs` filters out the inaccessible projects on the database-level, hence the API returns a 404 here. This returned a 403 before due to a bug pointed out above.